src/porepy/models/solution_strategy.py

```diff
@@ -437,13 +438,14 @@ def after_nonlinear_convergence(
         self.save_data_time_step()
 
     def after_nonlinear_failure(
-        self, solution: np.ndarray, errors: float, iteration_counter: int
+        self, solution: np.ndarray, errors: dict, iteration_counter: int
```

Comment:
I don't like changing the signature based on the needs identified in this particular PR. I appreciate the effort to make it more flexible by using a dict instead of, say, having the parameters errors_incr and errors_res. However, I think passing information between methods in this way is tricky when we can't predict what information may be required by future/tailor made versions of the methods. What about having a single object (dictionary, tailored class) for the current state of solver-relevant information? Perhaps simply as part of the (Non)LinearSolver object?

Reply:
If I understood correctly: The "errors" dictionary should possibly be an attribute of the NewtonSolver class? That does seem like a better solution. In that case though, wouldn't you still have to change the signature of the after_nonlinear_convergence/failure methods? At least if you would like to have both the residual and increment errors as input.

I can also mention that the previous "errors" object is a subset of the new "errors" object. The previous errors is a list, and the new errors is a dictionary of lists, so no information is lost. Related to this, is the current signature of after_nonlinear_convergence/failure wrong, because errors is a list and not a float?

Reply:
This discussion should now be resolved due the to introduction of the new `pp.SolverStatistics` object which is meant to be a medium of "whatever" is required. With internal tracking of these quantities, the signatures now changed again, as moving around data is not required anymore.